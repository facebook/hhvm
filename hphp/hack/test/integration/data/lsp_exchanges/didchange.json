[
{
  "jsonrpc": "2.0",
  "id": 1,
  "method": "initialize",
  "params":
    {
      "initializationOptions":{},
      "processId":null,
      "rootPath":"${root_path}",
      "capabilities":
        {
          "window":
            {
              "progress":{},
              "actionRequired":{}
          }
        }
    }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didOpen",
  "params": {
     "textDocument": {
       "uri":"${php_file_uri}",
       "languageId": "hack",
       "version":1,
       "text":"${php_file}"
     }
   }
},
{
  "jsonrpc": "2.0",
  "method": "textDocument/didChange",
  "params": {
      "textDocument": {
        "uri":"${php_file_uri}"
       },
       "contentChanges": [{
         "range": {
           "start": {"line":7, "character":11},
           "end": {"line":7, "character":11}
         },
         "text":"a"
       }]
   }
}
]
