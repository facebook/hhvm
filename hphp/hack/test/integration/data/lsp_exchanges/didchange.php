<?hh  //strict

function a_didchange(): int {
  return b_didchange();
}

function b_didchange(): int {
  return 42;
}
