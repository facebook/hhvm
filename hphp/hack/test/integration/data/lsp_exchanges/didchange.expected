[
  {
    "jsonrpc": "2.0",
    "id": 1,
    "result": {
      "capabilities": {
        "textDocumentSync": {
          "openClose": true,
          "change": 2,
          "willSave": false,
          "willSaveWaitUntil": false,
          "save": {
            "includeText": false
          }
        },
        "hoverProvider": true,
        "completionProvider": {
          "resolveProvider": true,
          "triggerCharacters": [
            "$",
            ">",
            "\\",
            ":",
            "<"
          ]
        },
        "definitionProvider": true,
        "referencesProvider": true,
        "documentHighlightProvider": true,
        "documentSymbolProvider": true,
        "workspaceSymbolProvider": true,
        "codeActionProvider": false,
        "documentFormattingProvider": true,
        "documentRangeFormattingProvider": true,
        "renameProvider": false,
        "typeCoverageProvider": true,
        "rageProvider": true
      }
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "telemetry/event",
    "params": {
      "type": 4,
      "message": "Version in hhconfig=[NoVersion]"
    }
  },
  {
    "jsonrpc": "2.0",
    "method": "textDocument/publishDiagnostics",
    "params": {
      "uri": "${php_file_uri}",
      "diagnostics": [
        {
          "range": {
            "start": {
              "line": 7,
              "character": 11
            },
            "end": {
              "line": 7,
              "character": 12
            }
          },
          "severity": 1,
          "code": 1002,
          "source": "Hack",
          "message": "Expected ;",
          "relatedLocations": []
        }
      ]
    }
  }
]
